ext/intl/spoofchecker/spoofchecker_main.c

```diff
@@ -167,3 +165,74 @@ PHP_METHOD(Spoofchecker, setRestrictionLevel)
 }
 /* }}} */
 #endif
+
+PHP_METHOD(Spoofchecker, setAllowedChars)
+{
+	zend_string *pattern;
+	UChar *upattern = NULL;
+	int32_t upattern_len = 0;
+	zend_long pattern_option = 0;
+	SPOOFCHECKER_METHOD_INIT_VARS;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_STR(pattern)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(pattern_option)
+	ZEND_PARSE_PARAMETERS_END();
+	SPOOFCHECKER_METHOD_FETCH_OBJECT;
+
+	if (ZSTR_LEN(pattern) > INT32_MAX) {
+		zend_argument_value_error(1, "is larger than " ZEND_LONG_FMT, INT32_MAX);
```

Comment:
This is the most common wording:

```suggestion
        zend_argument_value_error(1, "must be less than or equal to " ZEND_LONG_FMT " bytes", INT32_MAX);
```